src/model_config_utils.cc

```diff
@@ -1712,6 +1712,26 @@ ValidateInstanceGroup(
   return Status::Success;
 }
 
+Status
+ValidateNonLinearFormatIO(
+    const inference::ModelInput& io, const std::string& platform, bool is_input)
+{
+  if ((platform != kTensorRTPlanPlatform) && io.is_non_linear_format_io()) {
```

Comment:
Add this in the beginning to drop repeated checks for whether io.is_non_linear_format_io() below.
```
if(!io.is_non_linear_format_io()) {
   // Nothing to validate as the tensor is not non-linear format. 
   return Status::Success;
}
```